hedera-node/src/main/java/com/hedera/services/state/logic/SigsAndPayerKeyScreen.java

```diff
@@ -43,34 +45,31 @@ public class SigsAndPayerKeyScreen {
 	private final Rationalization rationalization;
 	private final PayerSigValidity payerSigValidity;
 	private final MiscSpeedometers speedometers;
-	private final TransactionContext txnCtx;
 	private final BiPredicate<JKey, TransactionSignature> validityTest;
 
 	@Inject
 	public SigsAndPayerKeyScreen(
 			Rationalization rationalization,
 			PayerSigValidity payerSigValidity,
-			TransactionContext txnCtx,
 			MiscSpeedometers speedometers,
 			BiPredicate<JKey, TransactionSignature> validityTest
 	) {
-		this.txnCtx = txnCtx;
 		this.validityTest = validityTest;
 		this.speedometers = speedometers;
 		this.rationalization = rationalization;
 		this.payerSigValidity = payerSigValidity;
 	}
 
-	public ResponseCodeEnum applyTo(SwirldsTxnAccessor accessor) {
+	public ResponseCodeEnum applyTo(SwirldsTxnAccessor accessor, Optional<TransactionContext> txnCtx) {
```

Comment:
We should try to avoid _any_ unnecessary object creations in the `handleTransaction` critical path---even `Optional`'s. So I'd suggest `@Nullable TransactionContext txnCtx` here.

Reply:
I reverted this changes since it's no longer needed